profile/README.md

```diff
@@ -1 +1,74 @@
-# Profile
\ No newline at end of file
+# Profile
+
+## Scopes concept
+
+"Scope-specific" data should be saved using the sub-keypair which makes the most sense for the content.
+
+### Examples for scopes
+
+SkyID has access to every scope and manages the master key
+`skysocial`: Social graph (all `userRelations` objects)
+`skyfeed`: All public data organized in a "feed" (uses `feedPage` objects) [posts, comments, reposts, reactions]
+`skylive`: Livestreams
+`skymessage`: Chat messages
+`something with storage`: Private file storage
+`skapp`: skapp.io specific developer and skapp data
+
+scope names MUST be valid HNS domains, because they link to their definition and description.
```

Comment:
Is it problematic that someone's social graph is reliant on a privately owned hns domain?
Should this be handled by giving them over to the foundation to provide security to the users? Or am I just misunderstanding how the domains function?

Reply:
The entity owning the domain has no control over any data stored with it. The idea to use HNS domains is only there, because it would allow for example the domain `skysocial` to contain a TXT record like `skystandards-scope=SKYLINK` with metadata about this scope like an icon, description and some more information which can be shown in the permissions dialog in SkyID and other Auth-Skapps

Reply:
The advantage with this concept is that new scopes can be added by developers at any time, without the need to define them in a central location and still having a nice icon and description for their scope when requested in an application

Reply:
Is there a way we can make these more generic? With this structure it means new developers are going to need to know every Skapp HNS name that they want user data for right? If that is the case it breaks part of the magic of the data being interoperable between skapps doesn't it?

If I go and make skyfeed2, I would want to be able to read `feed` from the users profile and likewise update `feed`. Then a user can come to skyfeed2 and immediately see all their posts and their friends posts that were made on skyfeed. Additionally, if skyfeed2 only has one user, that user should still be able to see all the stuff that is happening on skyfeed and vice versa if both skyfeed and skyfeed2 are using the same `feed` model. 

I don't think it is bad thing for developers to be forced to come to some community managed repo to suggest changes to things like `feed` if they have suggestions to improve it. But if we are keeping data silo'd with the users profile, that is only partially better than it being silo'd in different companies. 

The developers are then responsible for constantly searching for new Skapps that might be the same data type so they can piggy back on that network effect. 

Maybe I'm misunderstanding the intention here.

Reply:
Maybe the solution is for the scopes to be move one level down and grouped by Skapp type.  Skapps could belong to multiple types, but then developers could look for a type of skapp instead of a specific skapp.

```json
{
  "scopetypes": {
    "feed": {
       "skyfeed": {},
       "skyfeed2": {}
     },
    "streaming": {
       "skylive":{},
       "skylive2":{}
     }
   }
}
```

Reply:
then developers can still add scope at anytime and are only bottlenecked if they don't think their skapp fits in a predefined scope type. 

This way new developers can also immediately see all `feed` related scopes that the user has been involved with.

Reply:
Any app can request access to the scope `skyfeed` and even multiple scopes. (As I tried to explain in the "Examples how Skapps can use this scopes" section")
The apps `skyfeed` and `skyfeed2` can both request access to the scope and get the exact same private key.

But I'm also ok with using more generic types like "feed" and "live" and managing them in this repo, but this would break the ease of creating new scopes.

Reply:
I guess the comment about the scopes needing to be related to HNS domains confused me.  That makes it seem just like a list of skapps not scopes. 

That makes it seem hard for developers to know what is available. 

If I’m understanding it correctly, scopes should be how developers find the PubKey of a user that relates to the SkyDB entry that contains data that aligns with that scope. Am I understanding that right?  

Maybe the names are just throwing me off. SkyFeed hns would correspond to a scope schema? Or is SkyFeed in this context referring to the Skapp?

Reply:
> If I’m understanding it correctly, scopes should be how developers find the PubKey of a user that relates to the SkyDB entry that contains data that aligns with that scope. Am I understanding that right?

yes

> Maybe the names are just throwing me off. SkyFeed hns would correspond to a scope schema? Or is SkyFeed in this context referring to the Skapp?

`skyfeed` in this context would refer to a scope, but you're right that it could be confusing, so maybe it's better to just define all possible scopes in this repo (and use more generic names)